tests/MongoDB.Driver.Core.Tests/Core/Operations/AsyncCursorTests.cs

```diff
@@ -329,34 +330,96 @@ public void Session_reference_count_should_be_decremented_as_soon_as_possible(in
             var documents = Enumerable.Range(1, collectionSize).Select(n => new BsonDocument("_id", n));
             Insert(documents);
 
-            var findOperation = new FindOperation<BsonDocument>(_collectionNamespace, BsonDocumentSerializer.Instance, _messageEncoderSettings)
+            var cancellationToken = CancellationToken.None;
+            using (var binding = new ReadPreferenceBinding(CoreTestConfiguration.Cluster, ReadPreference.Primary, _session.Fork()))
+            using (var channelSource = binding.GetReadChannelSource(cancellationToken))
+            using (var channel = channelSource.GetChannel(cancellationToken))
             {
-                BatchSize = batchSize
-            };
+                var query = new BsonDocument();
+                long cursorId;
+                var firstBatch = GetFirstBatch(channel, query, batchSize, cancellationToken, out cursorId);
```

Comment:
GetFirstBatch will decide whether to use a "find" command or OP_QUERY depending on the server version.